library/std/src/sys/pal/unix/fs.rs

```diff
@@ -1753,8 +1753,8 @@ pub fn link(original: &Path, link: &Path) -> io::Result<()> {
                     // Android has `linkat` on newer versions, but we happen to know `link`
                     // always has the correct behavior, so it's here as well.
                     cvt(unsafe { libc::link(original.as_ptr(), link.as_ptr()) })?;
-                } else if #[cfg(any(target_os = "macos", target_os = "solaris"))] {
-                    // MacOS (<=10.9) and Solaris 10 lack support for linkat while newer
+                } else if #[cfg(any(target_os = "macos"))] {
+                    // MacOS (<=10.9) lacks support for linkat while newer
```

Comment:
It looks like our [minimum supported](https://doc.rust-lang.org/nightly/rustc/platform-support.html) macOS version is 10.12 so this branch can be removed entirely.